app/scripts/modules/core/pipeline/config/pipelineConfigProvider.js

```diff
@@ -59,10 +59,45 @@ module.exports = angular.module('spinnaker.core.pipeline.config.configProvider',
       return angular.copy(stageTypes);
     }
 
-    function getConfigurableStageTypes() {
-      return getStageTypes().filter(function(stageType) {
-        return !stageType.synthetic && !stageType.provides;
+    function getConfigurableStageTypes(providers) {
+      let allStageTypes = getStageTypes();
```

Comment:
Yeah, it is initialized to `[]`, and when types register, they are `push`ed into that array.